src/models/appmodel.ts

```diff
@@ -1,6 +1,6 @@
 import { observable, action, computed } from "mobx";
-import { IAppModel } from "./i_appmodel"
-
+import { IAppModel } from "./i_appmodel";
+import { ComboboxItem } from "shared/components/Combobox";
```

Comment:
Remove this line.  The appmodel must never know about ComboboxItem- this breaks the MVVM abstraction.

Reply:
It's a Type. We aren't using the `types` folder so I put this type to the Combobox component.
I need this type to mark that some property in this model will have a ComboboxItem type

We can add a Suffix Type after name eg. ComboboxItemType etc.

in case of the `types` folder, we wouldn't import this type

Reply:
OK, leave it in and I will show what I mean during refactoring after this PR.

Reply:
Sorry Erick, I really don't understand how can I change it without `types` folder.
In TS as like in C# I need to assign to the variable\property some type.
This type is defined in Combobox definition file (like you asked me before) and to let the Compiler know that i_appmodel property selectedItem should be a type of SelectedItem I need to import it directly.
Here is the screen
![image](https://user-images.githubusercontent.com/9897782/69911854-4a8a9700-142a-11ea-9837-4c72438bc3e1.png)


Looking forward to see your solution

Reply:
It's OK.  I'm ready to play with the code at this point and it will be easier to show what I mean by making my own changes.  It's also quite possible I'm missing something and I'll learn it by actually trying to change the code.